server/service/activities.go

```diff
@@ -45,3 +46,56 @@ func (svc *Service) ListActivities(ctx context.Context, opt fleet.ListActivities
 func (svc *Service) NewActivity(ctx context.Context, user *fleet.User, activity fleet.ActivityDetails) error {
 	return svc.ds.NewActivity(ctx, user, activity)
 }
+
+////////////////////////////////////////////////////////////////////////////////
+// List host upcoming activities
+////////////////////////////////////////////////////////////////////////////////
+
+type listHostUpcomingActivitiesRequest struct {
+	HostID      uint              `url:"id"`
+	ListOptions fleet.ListOptions `url:"list_options"`
+}
+
+func listHostUpcomingActivitiesEndpoint(ctx context.Context, request interface{}, svc fleet.Service) (errorer, error) {
+	req := request.(*listHostUpcomingActivitiesRequest)
+	acts, meta, err := svc.ListHostUpcomingActivities(ctx, req.HostID, req.ListOptions)
+	if err != nil {
+		return listActivitiesResponse{Err: err}, nil
+	}
+
+	return listActivitiesResponse{Meta: meta, Activities: acts}, nil
+}
+
+// ListHostUpcomingActivities returns a slice of upcoming activities for the
+// specified host.
+func (svc *Service) ListHostUpcomingActivities(ctx context.Context, hostID uint, opt fleet.ListOptions) ([]*fleet.Activity, *fleet.PaginationMetadata, error) {
+	// TODO(mna): should that require "read activity" or "read host" permission?
+	// Since this is specific activities for a host, I think it should be "read
+	// host" permission. To confirm with product.
```

Comment:
Forgot to ask confirmation for this one, I think authorizing on "read host" makes more sense (as it will properly refuse if the user is not in the host's team and doesn't otherwise have global read access).

Reply:
@jahzielv just want to bring your attention to this, if we're checking with product, do you mind adding a link here? if you've already got an answer, could you update this? Thanks!

Reply:
I asked here: https://github.com/fleetdm/fleet/issues/15958#issuecomment-1904027887

Reply:
This was confirmed (see reply to linked question), `read host` is correct, the TODO comment should be removed.